s_vae/models/backbone/unif_on_sphere.py

```diff
@@ -0,0 +1,50 @@
+import torch 
+import math
+from torch.distributions.distribution import Distribution
+
+class UnifOnSphere(Distribution):
+    """
+    Creates a uniform distribution on a m-dimensional hypersphere
+
+    Args:
+    ndim (Number, Tensor): Dimension of the hypersphere   
+    device "cpu"/"gpu": device to have the torch.tensor stored on 
+    """
+
+    has_rsample = False
+    support = torch.distributions.constraints.real
+    _mean_carrier_measure = 0
```

Comment:
Are these lines doing anything?

Reply:
`has_rsample = False` tells the Distribution class that there is no sampling of transformed density
`support` I don't know what does exactly, but I imagine it wants to know what the support of the distribution is to calculate distribution related stuff. 

`mean_carrier_measure` is an exponential family thing.

But yeah, I just saw in the tutorial that they did this, so I assumed it was needed/useful for the Distribution class.

Reply:
Actually just primarily looked at how `torch` implemented the Multivariate Normal.

Reply:
https://pytorch.org/docs/stable/_modules/torch/distributions/multivariate_normal.html#MultivariateNormal